### PR TITLE
Feat/custom tag

### PR DESCRIPTION
**Pull request by 2976151305**
feat: Optimize the presentation of custom label components in code blocks

### CODE REVIEW

File: components/chat/completion.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { IChatDialogueMessageSchema } from '@/types/chat';
 import classNames from 'classnames';
 import { Empty, Modal, message, Tooltip } from 'antd';
 import { renderModelIcon } from './header/model-selector';
-import loadsh from 'lodash';
+import _ from 'lodash';
```
Comment (Aralhi) [inline components/chat/completion.tsx:11]: import {cloneDeep} from 'lodash'

File: components/chat/chat-content/config.tsx [hunk 2]
```diff
@@ -8,18 +8,42 @@ import { Reference } from '@/types/chat';
...
+      return '';
+    });
+    return acc;
+  }, []);
+  return { context, matchValues };
+}
+
 const basicComponents: MarkdownComponent = {
   code({ inline, node, className, children, style, ...props }) {
+    const { context, matchValues } = matchCustomeTagValues(Array.isArray(children) ? children.join('\n') : children);
```
Comment (Aralhi) [inline components/chat/chat-content/config.tsx:31]: add a comment

File: components/chat/chat-content/config.tsx [hunk 3]
```diff
@@ -8,18 +8,42 @@ import { Reference } from '@/types/chat';
...
+    return (
+      <>
+        {!inline && match ? (
+          <CodePreview code={context} language={match?.[1] ?? 'javascript'} />
+        ) : (
+          <code {...props} style={style} className="px-[6px] py-[2px] rounded bg-gray-700 text-gray-100 dark:bg-gray-100 dark:text-gray-800 text-sm">
+            {children}
+          </code>
+        )}
+        <ReactMarkdown components={markdownComponents} rehypePlugins={[rehypeRaw]}>
```
Comment (Aralhi) [inline components/chat/chat-content/config.tsx:43]: maybe we can add more common plugins?

File: components/chat/chat-content/config.tsx (hunk 2 above)
Comment (2976151305) [reply components/chat/chat-content/config.tsx:36]: done

File: components/chat/completion.tsx (hunk 1 above)
Comment (2976151305) [reply components/chat/completion.tsx]: done

**Review by Aralhi — CHANGES_REQUESTED** [review]
done

**Pull request merged**